src/containers/Header/index.tsx

```diff
@@ -1,48 +1,40 @@
 import { ConnectButton } from '@rainbow-me/rainbowkit';
-import styled from 'styled-components';
+import { styled } from '@mui/material/styles';
+import { IconButton } from '@mui/material';
+import LightModeIcon from '@mui/icons-material/LightMode';
+import DarkModeIcon from '@mui/icons-material/DarkMode';
 
-import { useStateContext } from '~/hooks';
-import { THEME_KEY } from '~/utils';
+import { useCustomTheme } from '~/hooks/useTheme';
 
 export const Header = () => {
-  const { setTheme, theme } = useStateContext();
-
-  const handleThemeChange = () => {
-    if (theme === 'light') {
-      localStorage.setItem(THEME_KEY, 'dark');
-      setTheme('dark');
-    } else {
-      localStorage.setItem(THEME_KEY, 'light');
-      setTheme('light');
-    }
-  };
+  const { changeTheme, theme } = useCustomTheme();
 
   return (
-    <HeaderContainer>
-      <h1>Logo</h1>
-      <ThemeButton onClick={handleThemeChange}>{theme === 'dark' ? '🌞' : '🌕'}</ThemeButton>
+    <StyledHeader>
+      <Logo>Logo</Logo>
+      <IconButton onClick={changeTheme}>{theme === 'dark' ? <LightModeIcon /> : <DarkModeIcon />}</IconButton>
       <ConnectButton />
-    </HeaderContainer>
+    </StyledHeader>
   );
 };
 
-const HeaderContainer = styled.div`
-  display: flex;
-  height: 8rem;
-  padding: 0 8rem;
-  align-items: center;
-  justify-content: space-between;
-  background-color: ${({ theme }) => theme.headerBackground};
-  border-bottom: ${({ theme }) => theme.border};
-  width: 100%;
-  max-width: 100vw;
-  z-index: 100;
-`;
+//Styles
+const StyledHeader = styled('header')(() => {
+  const { currentTheme } = useCustomTheme();
+  return {
+    display: 'flex',
+    height: '8rem',
+    padding: '0 8rem',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    backgroundColor: currentTheme.backgroundSecondary,
+    width: '100vw',
+    zIndex: 100,
```

Comment:
put on constants in another pr <3

Reply:
https://github.com/defi-wonderland/web3-react-boilerplate/pull/8/commits/a4a80bcb3d05cb213b045a2f40932f8ae160615c#diff-31f5dea1a137fef36b405b82a17c26ff39f1991fec1f17f2693bd4e7d3ed9b13